crates/services/txpool_v2/src/ports.rs

```diff
@@ -0,0 +1,79 @@
+use fuel_core_storage::Result as StorageResult;
+use fuel_core_types::{
+    entities::{
+        coins::coin::CompressedCoin,
+        relayer::message::Message,
+    },
+    fuel_tx::{
+        BlobId,
+        Bytes32,
+        ContractId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+    fuel_vm::interpreter::Memory,
+    tai64::Tai64,
+};
+
+use crate::{
+    error::Error,
+    GasPrice,
+};
+
+/// Provides an atomic view of the storage at the latest height at
+/// the moment of view instantiation. All modifications to the storage
+/// after this point will not affect the view.
+pub trait AtomicView: Send + Sync {
```

Comment:
Oh. I was thinking we were using the trait from `fuel_storage`. Maybe this is okay. I'm not sure I understand how it's being used yet though.

Reply:
The goal is to have a way to get a read only access, most updated possible, to the database and we interact with the database with `TxPoolDb`

Reply:
I'm also not sure about this trait definition? 
Wouldn't 
```rust
pub use fuel_storage::transactional::AtomicView
``` 
suffice if you want to expose the AtomicVIew trait through this module? 

Or is the interface supposed to evolve differently from `fuel_storage::transactional::AtomicView`?

Reply:
I can just reexport yes. Changed.